### PR TITLE
tests: Work around race condition when detecting bad certificate.

### PR DESCRIPTION
**Pull request by kacf**
If a bad certificate is detected, which is what we are testing, the
Golang request processing (http.Client.Do) will close the connection.
Sometimes, but not always, this will lead to a "connection closed"
error instead of a "bad certificate" error, the latter of which we are
looking for in the string. It appears that this is outside of our
control, since our server is not torn down until after the check has
been made.

Work around this by repeating the test multiple times using
assert.Eventually.

Changelog: None

Signed-off-by: Kristian Amlie <kristian.amlie@northern.tech>

### CODE REVIEW

**Comment by kacf** [comment]
Cancelling mender-qa pipeline. This only affects unit tests.

File: app/mender_test.go [hunk 1]
```diff
@@ -1243,15 +1254,24 @@ func TestMutualTLSClientConnection(t *testing.T) {
...
+				err = mender.UploadLog(
+					&datastore.UpdateInfo{
+						ID: "foobar",
+					},
+					logs,
+				)
+				t2.failed = false
+				test.assertFunc(t2, err, srv.Log.Logs)
+				return !t2.failed
+			}, 10*time.Second, 500*time.Millisecond)
```
Comment (merlin-northern) [inline app/mender_test.go:1269]: maybe we could define those timeouts somewhere above as vars or in the test case.

**Review by merlin-northern — APPROVED** [review]
lgtm

File: app/mender_test.go (hunk 1 above)
Comment (kacf) [reply app/mender_test.go]: Done!

**Review by oleorhagen — APPROVED** [review]

File: app/mender_test.go (hunk 1 above)
Comment (merlin-northern) [reply app/mender_test.go]: lovely!

**Pull request merged**